mlrun/model.py

```diff
@@ -1514,26 +1524,45 @@ def ui_url(self) -> str:
 
     @property
     def outputs(self):
-        """return a dict of outputs, result values and artifact uris"""
+        """
+        Return a dictionary of outputs, including result values and artifact URIs.
+
+        This method ensures that only artifacts with deterministic tags (excluding "latest") are included.
+
+        :return: Dictionary
+        """
         outputs = {}
         self._outputs_wait_for_completion()
         if self.status.results:
-            outputs = {k: v for k, v in self.status.results.items()}
+            outputs.update(self.status.results)
         if self.status.artifacts:
-            for a in self.status.artifacts:
-                key = a["metadata"]["key"]
-                outputs[key] = get_artifact_target(a, self.metadata.project)
+            # Exclude artifacts tagged as 'latest' to ensure a deterministic tag is returned for the artifact
+            outputs.update(
+                {
+                    artifact["metadata"]["key"]: get_artifact_target(
+                        artifact, self.metadata.project
+                    )
+                    for artifact in self.status.artifacts
+                    if artifact["metadata"].get("tag", None) != "latest"
+                }
+            )
```

Comment:
But what if we log artifacts without an explicit tag?
They will automatically be tagged as `latest`, and will be excluded in the output.